authors/apps/articles/views.py

```diff
@@ -310,3 +311,89 @@ def get(self, request, slug):
+    """This class creates a new like or dislike"""
+    permission_classes = (IsAuthenticated,)
+    serializer_class = BookmarkSerializer
+
+    def post(self, request, slug):
+        """Creates a like"""
+        data = request.data
+        article = Article.objects.filter(
+            slug=slug, published=True, activated=True).first()
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+
+        bookmark = Bookmark.objects.filter(
+            user_id=request.user.pk, article_id=article.id).first()
+        if bookmark is not None:
+            favorite_found = {
+                "detail": "Article already bookmarked."
+            }
+            return Response(data=favorite_found,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        data['article_id'] = article.id
+        data['user_id'] = request.user.pk
+        serializer = self.get_serializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        data['detail'] = 'Article added to bookmarks.'
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, *args, **kwargs):
+        article = Article.objects.filter(
+            slug=kwargs['slug'], published=True, activated=True).first()
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+        favorite = Bookmark.objects.filter(
+            article_id=article.id, user_id=request.user.pk).first()
+        if favorite:
+            message = {"detail": "Article removed from bookmarked articles"}
+            favorite.delete()
+            return Response(data=message, status=status.HTTP_204_NO_CONTENT)
+        not_found = {"detail": "Article not bookmarked"}
+        return Response(not_found, status.HTTP_404_NOT_FOUND)
+
+    def get(self, request, *args, **kwargs):
+        """"""
+        article = Article.objects.filter(
+            slug=kwargs['slug'], published=True, activated=True).first()
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+        favorite = Bookmark.objects.filter(
+            article_id=article.id, user_id=request.user.pk).first()
+        if favorite:
+            serializer = self.serializer_class(favorite)
+            return Response(data=serializer.data, status=status.HTTP_200_OK)
+        not_found = {"detail": "Article not bookmarked"}
+        return Response(not_found, status.HTTP_404_NOT_FOUND)
+
+
+class GetUserBookmarksView(APIView):
+    """Gets all articles' likes and dislikes"""
```

Comment:
@ElMonstro, kindly work on your docstrings.

Reply:
Ok.. I will work on that